src/lib/mappers/charge-element.js

```diff
@@ -18,55 +19,96 @@ const timeLimitedDateMapper = (startDate, endDate) =>
 const dbToModelMapper = createMapper()
   .map('chargeElementId').to('id')
   .copy(
+    'scheme',
     'source',
     'season',
     'loss',
-    'authorisedAnnualQuantity',
-    'billableAnnualQuantity',
     'description',
-    'chargeVersionId',
-    'isSection127AgreementEnabled'
-  )
-  .map().to('abstractionPeriod', abstractionPeriodMapper.dbToModel)
-  .map('factorsOverridden').to('isFactorsOverridden')
-  .map('purposePrimary').to('purposePrimary', purposePrimaryMapper.dbToModel)
-  .map('purposeSecondary').to('purposeSecondary', purposeSecondaryMapper.dbToModel)
-  .map('purposeUse').to('purposeUse', purposeUseMapper.dbToModel)
-  .map(['timeLimitedStartDate', 'timeLimitedEndDate']).to('timeLimitedPeriod', timeLimitedDateMapper, { mapNull: false });
+    'chargeVersionId'
+  );
 
 /**
  * Creates a ChargeElement instance given a row of charge element data
- * @param {Object} chargeElementRow - charge element row from the charge processor
+ * @param {Object} row - charge element row from the charge processor
  * @return {ChargeElement}
  */
-const dbToModel = row =>
-  helpers.createModel(ChargeElement, camelCaseKeys(row), dbToModelMapper);
-
+const dbToModel = row => {
+  if (row.scheme === 'sroc') {
+    return helpers.createModel(ChargeElement, camelCaseKeys(row), dbToModelMapper
+      .copy(
+        'volume',
+        'waterModel',
+        'isRestrictedSource',
+        'eiucRegion'
+      )
+      .map('chargePurposes').to('chargePurposes', chargePurposes => chargePurposes.map(chargePurposeMapper.dbToModel))
+      .map('chargeCategory').to('chargeCategory', chargeCategoryMapper.dbToModel)
+    );
+  } else {
+    return helpers.createModel(ChargeElement, camelCaseKeys(row), dbToModelMapper
+      .copy(
+        'authorisedAnnualQuantity',
+        'billableAnnualQuantity',
+        'isSection127AgreementEnabled',
+        'isSection127AgreementEnabled'
+      )
+      .map().to('abstractionPeriod', abstractionPeriodMapper.dbToModel)
```

Comment:
I think I mentioned this in a previous review: `.map()` doesn't feel right. Is there any point in mapping an undefined value?

Reply:
Thanks I missed this one from the previous work thanks